script.js

```diff
@@ -1,12 +1,440 @@
-//You can edit ALL of the code here
+//Get DOM elements
+const rootHtml = document.getElementById("root"); // Get the root element where we'll add the episode cards
+const episodeSearchInputHtml = document.getElementById("episode-search-input");
+const showSearchInputHtml = document.getElementById("show-search-input");
+const episodeSearchResultDisplayHtml = document.querySelector(".display-episode-search");
+const showSearchResultDisplayHtml = document.querySelector(".display-show-search");
+const episodeSelectHtml = document.querySelector("#episode-select-html");
+const showSelectHtml = document.querySelector("#show-select-html");
+const showListingButton = document.querySelector("#show-listing-button");
+
+// Global variables
+let allEpisodes;
+let allShows;
+
+window.onload = setup;
+
+// Fetch episodes function
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  allShows = getAllShows().sort((a, b) => {
+    return a.name.localeCompare(b.name); // sort shows into alphabeltical order by comparing string, not numbers
+  });
+
+  buildShowDropdownList(allShows);
+  makePageForShows(allShows);
+}
+
+function fetchShowEpisodes(showId) {
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`) // returns a promise and it is pending
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json(); // returns the promise as fulfilled and gives us a response object which we pass into the callback function
+      } else {
+        throw new Error(
+          `Encountered something unexpected: ${response.status} ${response.statusText}`
+        );
+      }
+    })
+    .then((data) => {
+      allEpisodes = data;
+      showSearchInputHtml.value = "";
+      makePageForEpisodes(allEpisodes);
+      buildEpisodeDropdownList(allEpisodes);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
 }
 
+// level 100
+
+/*
+  If the user clicks show select then the show select will disappear
+  If the user clicks the see show listing button then the episodes will disappear and the shows will appear also the episode list should not be shown initially
+*/
+
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  episodeSelectHtml.style.display = "inline"; // make the episode dropdown list visible
+  showSelectHtml.style.display = "none"; // hide the episode dropdown list
+
+  episodeSearchInputHtml.style.display = "inline"; // make the show searchbar visible
+  showSearchInputHtml.style.display = "none"; // hide episode searchbar from show page
+
+  episodeSearchResultDisplayHtml.style.display = "inline";
+  showSearchResultDisplayHtml.style.display = "none"; // hide episode search result from show page
+
```

Comment:
It looks like you're doing the same thing 6 times which is very repetitive. Is there possibly a good word to possibly describe what we're doing. Is there a function name that we can use, using that word.

Reply:
Thank you very much for pointing this out! Is the word to display and hide called called "toggle"?

Reply:
Hi Afsha,
That is up to you. What I would say is that you should try it and see how it goes with "toggle()" and if you need any more help let myself or one of the team know :)

Reply:
I might be able to manage this one. There are a lot of other bug I need to fix. It feels overwhelming sometimes. Thank you for you so much for your kind offer!

Reply:
No problem! Trust me I am still learning as well so do not worry one bit. A bit of advice I can give in terms of feeling overwhelmed is to try and tell someone or write down al the problems you are having as you can tend to find the answer while you're speaking about the problem.

i hope this helps and let me know if you need anything else!

Reply:
Hey Faisal, 
I have made some changes to reduce some repetition by using a toggleVisibility function. 
For me there is nothing like speaking about the problem. It gives me clarity. However, rarely I get to speak about the coding problems. I have been making checklists about the all the coding related problems in a notebook and have made some progress. Thank you for the reminder!
I am currently struggling a few small assignments related to fetch and API and we are starting React. Would you be interested in to pair programming sometime?

Reply:
Hi Afsha,
I apologise for the late reply. I did not see this until today when I logged into Gituhub. Yes I would be happy to start pair programming. I am not super strong in programming as I have not done much since I finished my degree however, I still practice in my free time. I would be happy to do it. Let me know when you want to do it.

Reply:
Hey Faisal!
No worries about the delay. Before we start pair programming, it would be great if you could join our in-person training sessions on Saturdays. It'll give us a chance to meet face-to-face and discuss pair programming and our projects in more detail.
Let me know if you're interested, and I'll share the details of the upcoming Saturday session. You can also find the information in our Slack channel. Feel free to check it out.

Reply:
Hi Afsha,
Yes I will be more than happy to come in person for a session. I won't be able to attend tomorrow as I am away for the weekend however nest Saturday works perfectly for me. Could you please forward me the details and I will see you then!

Reply:
Tomorrow it will be remote. Next week it will probably be at Lego, which is exciting! Feel free to DM on slack or here if you need the address.